packages/core/src/memory/memory.ts

```diff
@@ -159,13 +159,17 @@ export abstract class MastraMemory extends MastraBase {
   protected parseMessages(messages: MessageType[]): CoreMessage[] {
     return messages.map(msg => ({
       ...msg,
-      content:
-        typeof msg.content === 'string' && (msg.content.startsWith('[') || msg.content.startsWith('{'))
-          ? JSON.parse((msg as MessageType).content as string)
-          : typeof msg.content === 'number'
-            ? String(msg.content)
-            : msg.content,
-    }));
+      content: (() => {
+        if (typeof msg.content === 'string' && (msg.content.startsWith('[') || msg.content.startsWith('{'))) {
+          try {
+            return JSON.parse(msg.content);
+          } catch (e) {
+            return msg.content;
+          }
+        }
+        return typeof msg.content === 'number' ? String(msg.content) : msg.content;
+      })()
```

Comment:
on second thought here, i dont like the func here.

Can we sort out what `content` should be above on line 163. Lets open up the function 


```
return messages.map(msg => {
 let content = msg.content
 if (typeof content === 'string' && (content.startsWith('[') || content.startsWith('{'))) {
  try {
    content = JSON.parse(content);
   } catch (e) {
       // swallow
   }
 } else if(typeof msg.content === 'number') {
    content = String(msg.content);
 }
  return {
      ...msg,
      content,
  }
})
```

something like this

Reply:
Thanks for review. I agree your comment, so fixed it :)